File: 01/main.py
import sys


def main():
    with open(sys.argv[1], "r") as f:
        x = [i.strip() for i in f.readlines()]
    elves = []
    elf = 0
    for item in x:
        if item != "":
            elf += int(item)
        else:
            elves.append(elf)
            elf = 0
    print(max(elves))
    print(sum(sorted(elves)[-3:]))


main()

File: 01/main.nim
import std/[os, strutils, algorithm, math]


proc main() =
  var
    x = paramStr(1).readFile.splitLines
    elves: seq[int]
    elf = 0
  for item in x:
    if item != "":
      elf += item.parseInt
    else:
      elves.add(elf)
      elf = 0
  echo max(elves)
  echo sum(sorted(elves, system.cmp)[^3..elves.high])


main()
